src/main/java/Duke.java

```diff
@@ -1,10 +1,89 @@
+import dukeexception.DukeException;
+import dukeexception.DukeUnknownInputException;
+
+/**
+ * Duke class uses Storage to load and unload data from and to files, TaskList to contain and operate on
+ * the lit of tasks, Ui to interact with the user, and Parser to decipher the user command.
+ */
 public class Duke {
+
+    private Storage storage;
+    private TaskList tasks;
+    private Ui ui;
+    private Parser parser;
+
+    /**
+     * Constructor of Duke.
+     * Attempts to load tasks from file, if fail start with empty list of tasks.
+     *
+     * @param filePath The path to the file to load tasks from.
+     */
+    public Duke(String filePath) {
+        ui = new Ui();
+        parser = new Parser();
+        storage = new Storage(filePath);
+        try {
+            tasks = new TaskList(storage.load());
+        } catch (DukeException e) {
+            ui.showError(e);
+            tasks = new TaskList();
+        }
+    }
+
+    /**
+     * A continuous loop of getting the user command, parsing to make sense of it and executing the command.
+     * Stops when the user command is "bye.
+     * @throws DukeException for any exceptions that occur during this run.
+     */
+    public void run() {
+        ui.showHello();
+        while (true) {
+            String command = ui.readCommand();
+            if (command.equals("bye")) {
+                ui.showBye();
+                break;
+            }
+            String commandType = parser.getCommandType(command);
+            try {
+                switch (commandType) {
```

Comment:
I really liked the proper indentation of the switch case statement. Even me myself did this wrongly. Good job!